### PR TITLE
improve JS Edge version detection, add support for Edge Chromium

### PR DESCRIPTION
**Pull request by johndoh**
currently the JS browser version detection for Edge is not working quite right as it gets confused by other browser engines mentioned in the user agent. eg:

`Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.102 Safari/537.36 Edge/18.18938`

is detected as version `537.36` rather than `18.18938`. This fixes that.

also in the Edge Chromium the user agent string has changed slightly to:

`Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3763.0 Safari/537.36 Edg/75.0.131.0`

so i've added support for that as well.

### CODE REVIEW

File: program/js/common.js [hunk 1]
```diff
@@ -77,8 +77,14 @@ function roundcube_browser()
   }
 
   if (!this.vendver) {
-    // common version strings
-    this.vendver = /(opera|opr|khtml|chrome|safari|applewebkit|msie)(\s|\/)([0-9\.]+)/.test(this.agent_lc) ? parseFloat(RegExp.$3) : 0;
```
Comment (alecpl) [inline program/js/common.js]: Wouldn't just `/(edge?|opera|...` work here?

**Review by alecpl — COMMENTED** [review]
Question is if we still want to detect Edge Chromium as Edge not Chrome ;)

File: program/js/common.js (hunk 1 above)
Comment (johndoh) [reply program/js/common.js]: I tired that but for me it always picks up the `applewebkit` hit. may be I'm doing something wrong

**Comment by johndoh** [comment]
> Question is if we still want to detect Edge Chromium as Edge not Chrome ;)

a fair question. i guess this is a similar case to how safari is handled. where safari is set in JS but the class `chrome` is added to the html. the same could be done for Edge

File: program/js/common.js (hunk 1 above)
Comment (alecpl) [reply program/js/common.js]: Ah, I think it takes the longest match not the first match.

Comment (alecpl) [reply program/js/common.js]: or first match looking at the input not the regexp

**Comment by johndoh** [comment]
Second attempt. I've rewritten the browser version code as I found errors in multiple browsers with the current code. new code is tested with latest user-agents from:

* IE10
* IE11
* Edge
* Edge Chromium
* Opera
* Konqueror
* Safari
* Chrome
* Firefox

I've also set the html class to `chrome` for Edge-Chromium as I think this makes the most sense for CSS.

**Pull request merged**